stdlib/os/__init__.pyi

```diff
@@ -367,6 +367,8 @@ def listdir(path: bytes) -> list[bytes]: ...
 def listdir(path: int) -> list[str]: ...
 @overload
 def listdir(path: PathLike[str]) -> list[str]: ...
+@overload
+def listdir(path: PathLike[bytes]) -> list[bytes]: ...
```

Comment:
It seems to me that the last two overloads can be combined with the first two, e.g. `listdir(path: str | PathLike[str] | None = ...) -> list[str]`. The separate overloads were introduced in #1894 (found with a mess of `git blame` and `git checkout`), and it shows no reason to not use a union.

Note that `_typeshed.StrPath` is the preferred way to spell `str | PathLike[str]`. You can use it if it works. I'm not sure, because `PathLike` is defined just above.

Reply:
@Akuli Good shout, I've updated to using `StrPath` and `BytesPath` and the tests don't seem to have any issue with this 😄